File: platform-101/03-model-compare.ts
import "dotenv/config";
import Anthropic from "@anthropic-ai/sdk";

const client = new Anthropic();

const prompt = "Explain Prompt coaching in two sentences";
const models = ["claude-haiku-4-5", "claude-sonnet-4-6", "claude-opus-4-8"];

for (const model of models) {
    const start = Date.now()

    const response = await client.messages.create({
        model,
        max_tokens: 300,
        messages: [{ role: "user", content: prompt }],
    });

    const ms = Date.now() - start
    console.log(`\n[${model}] ${ms}ms in=${response.usage.input_tokens} out=${response.usage.output_tokens}`);

    for (const block of response.content) {
        if (block.type === "text") {
            console.log(block.text);
        }
    }
}

File: platform-101-python/03_model_compare.py
import time

from dotenv import load_dotenv
load_dotenv()

import anthropic

client = anthropic.Anthropic()

prompt = "Explain Prompt coaching in two sentences"
models = ["claude-haiku-4-5", "claude-sonnet-4-6", "claude-opus-4-8"]

for model in models:
    start = time.monotonic()

    response = client.messages.create(
        model=model,
        max_tokens=300,
        messages=[{"role": "user", "content": prompt}],
    )

    ms = (time.monotonic() - start) * 1000
    print(f"\n[{model}] {ms:.0f}ms in={response.usage.input_tokens} out={response.usage.output_tokens}")

    for block in response.content:
        if block.type == "text":
            print(block.text)
